python/torch_mlir/dialects/torch/importer/jit_ir/csrc/module_builder.cpp

```diff
@@ -174,6 +175,24 @@ void ModuleBuilder::importModule(torch::jit::Module jitModule,
   mlirOperationSetAttributeByName(mlirModuleGetOperation(module),
                                   toMlirStringRef("torch.debug_module_name"),
                                   debugModuleNameAttr);
+
+  std::string globalDType;
+  switch (torch::tensors::get_default_scalar_type()) {
+  case at::ScalarType::Half:
+    globalDType = "torch.float16";
+    break;
+  case at::ScalarType::Double:
+    globalDType = "torch.float64";
+    break;
+  default:
+    globalDType = "torch.float32";
+    break;
+  };
+  auto moduleDType = mlirStringAttrGet(context, toMlirStringRef(globalDType));
+  mlirOperationSetAttributeByName(mlirModuleGetOperation(module),
```

Comment:
@silvasean, is there no way to set an MLIR Type as an attribute to avoid comparing strings in RefineTypes?

Reply:
You can use a TypeAttr -- construct via mlirTypeAttrGet.

Reply:
Is there a way to get a Type here? I don't think there's an MLIR context at this point.